### PR TITLE
fix: attribute error with listeners in cogs

### PR DESCRIPTION
**Pull request by Om1609**
## Summary
Fixes Attribute Error caused by #1957 when using listeners in cogs

## Information

- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [ ] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.
- [x] I have updated the changelog to include these changes.

### CODE REVIEW

**Review by Lulalaby — DISMISSED** [review]

**Review by Lulalaby — APPROVED** [review]

File: discord/client.py [hunk 1]
```diff
@@ -442,8 +442,11 @@ def dispatch(self, event: str, *args: Any, **kwargs: Any) -> None:
         # Schedule additional handlers registered with @listen
         for coro in self._event_handlers.get(method, []):
             self._schedule_event(coro, method, *args, **kwargs)
-            if coro._once:
-                once_listeners.append(coro)
+            try:
+                if coro._once:
+                    once_listeners.append(coro)
+            except AttributeError:
```
[the remark this answers is not in the document]
Comment (Om1609) [reply discord/client.py:448]: Sure

File: CHANGELOG.md [hunk 2]
```diff
@@ -48,6 +48,9 @@ These changes are available on the `master` branch, but have not yet been releas
   ([#1955](https://github.com/Pycord-Development/pycord/pull/1955))
 - Fixed `reason` being passed to wrong method in `guild.create_auto_moderation_rule`.
   ([#1960](https://github.com/Pycord-Development/pycord/pull/1960))
+- Fixed `AttributeError` caused by
+  [#1957](https://github.com/Pycord-Development/pycord/pull/1957) when using listeners
+  in cogs ([#1989](https://github.com/Pycord-Development/pycord/pull/1989))
```
Comment (BobDotCom) [inline CHANGELOG.md:53]: Please move this to the "Unreleased" section.

**Review by Lulalaby — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -31,6 +31,12 @@ These changes are available on the `master` branch, but have not yet been releas
 - Removed `@client.once()` in favour of `@client.listen(once=True)`.
   ([#1957](https://github.com/Pycord-Development/pycord/pull/1957))
 
+### Fixed
+
+- Fixed `AttributeError` caused by
+  [#1957](https://github.com/Pycord-Development/pycord/pull/1957) when using listeners
+  in cogs ([#1989](https://github.com/Pycord-Development/pycord/pull/1989))
```
Comment (BobDotCom) [inline CHANGELOG.md:38]: Minor change
```suggestion
  in cogs. ([#1989](https://github.com/Pycord-Development/pycord/pull/1989))
```

**Comment by Om1609** [comment]
Can this be merged now?

File: CHANGELOG.md (hunk 3 above)
Comment (Lulalaby) [reply CHANGELOG.md]: geez

**Review by Lulalaby — APPROVED** [review]

**Comment by Lulalaby** [comment]
idk ask bob

**Review by BobDotCom — APPROVED** [review]

**Pull request merged**